test_deposit_withdrawal failure

## Current Behavior
Red,
```
AssertionError: Wallet balance in neofs is not correct after deposit
assert 0.0 == 100
 +  where 0.0 = get_neofs_balance(&lt;neofs_testlib.env.env.NeoFSEnv object at 0x104ed6f90&gt;, &lt;neofs_testlib.cli.neofs_cli.cli.NeofsCli object at 0x105403a10&gt;, NodeWallet(path=PosixPath('/Users/runner/work/neofs-testcases/neofs-testcases/neofs-testcases/test-run-2025-01-17-01-08-51-876678/env_files/neofs-env-2025-01-17-01-54-23/deposit_withdrawal_test_wallet_kaphicjxvq'), address='NM9zXRPWkR4BrPBLC3ErrwHUQAFFTW9JcW', password='password', cli_config=None, neo_go_config=None))
```
https://rest.fs.neo.org/HXSaMJXk2g8C14ht8HSi7BBaiYZ1HeWh2xnWPGQCg4H6/789-1737080713/index.html#suites/81c17107ccbab4b783c7cba2aa651151/bd7a5d907217944c/

## Expected Behavior
Green. This looks like a test issue at this point. Maybe it doesn't wait enough.

## Possible Solution
Fix it.

## Regression
No.

## Your Environment
Node 0.44.0.

--- a/pytest_tests/tests/payment/test_deposit_withdrawal.py
+++ b/pytest_tests/tests/payment/test_deposit_withdrawal.py
@@ -1,9 +1,17 @@
-import time
-
 import allure
+from helpers.test_control import wait_for_success
 from helpers.wallet_helpers import get_neofs_balance, get_wallet_balance
 from neofs_testlib.env.env import NeoFSEnv, NodeWallet
 from neofs_testlib.utils import wallet as wallet_utils
+
+
+@allure.step("Wait for correct neofs balance")
+@wait_for_success(60, 5)
+def wait_for_correct_neofs_balance(
+    neofs_env: NeoFSEnv, wallet: NodeWallet, cli_wallet_config: str, compare_func: callable
+):
+    neofs_cli = neofs_env.neofs_cli(cli_wallet_config)
+    assert compare_func(get_neofs_balance(neofs_env, neofs_cli, wallet)), "Wallet balance in neofs is not correct"
 
 
 class TestDepositWithdrawal:
@@ -51,10 +59,7 @@ class TestDepositWithdrawal:
             assert get_wallet_balance(neofs_env, neo_go, wallet, neo_go_wallet_config) <= 900, (
                 "Wallet balance is not correct after deposit"
             )
-            neofs_cli = neofs_env.neofs_cli(cli_wallet_config)
-            assert get_neofs_balance(neofs_env, neofs_cli, wallet) == 100, (
-                "Wallet balance in neofs is not correct after deposit"
-            )
+            wait_for_correct_neofs_balance(neofs_env, wallet, cli_wallet_config, lambda balance: balance == 100)
 
         with allure.step("Withdraw some money back to the wallet"):
             neo_go.contract.invokefunction(
@@ -66,10 +71,7 @@ class TestDepositWithdrawal:
                 multisig_hash=f"{wallet.address}:Global",
                 force=True,
             )
-            time.sleep(10)
-            assert get_neofs_balance(neofs_env, neofs_cli, wallet) == 50, (
-                "Wallet balance in neofs is not correct after withdrawal"
-            )
+            wait_for_correct_neofs_balance(neofs_env, wallet, cli_wallet_config, lambda balance: balance == 50)
             assert get_wallet_balance(neofs_env, neo_go, wallet, neo_go_wallet_config) > 940, (
                 "Wallet balance is not correct after withdrawal"
             )

